File: app/src/main/java/wangdaye/com/geometricweather/remoteviews/config/DayWidgetConfigActivity.java
package wangdaye.com.geometricweather.remoteviews.config;

import android.os.Build;
import android.view.View;
import android.widget.RemoteViews;

import wangdaye.com.geometricweather.R;
import wangdaye.com.geometricweather.remoteviews.presenter.DayWidgetIMP;

/**
 * Day widget config activity.
 * */

public class DayWidgetConfigActivity extends AbstractWidgetConfigActivity {

    @Override
    public void initData() {
        super.initData();
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
            String[] widgetStyles = getResources().getStringArray(R.array.widget_styles);
            String[] widgetStyleValues = getResources().getStringArray(R.array.widget_style_values);

            this.viewTypeValueNow = "rectangle";
            this.viewTypes = new String[] {
                    widgetStyles[0],
                    widgetStyles[1],
                    widgetStyles[2],
                    widgetStyles[3],
                    widgetStyles[4],
                    widgetStyles[5],
                    widgetStyles[6],
                    widgetStyles[7]
            };
            this.viewTypeValues = new String[] {
                    widgetStyleValues[0],
                    widgetStyleValues[1],
                    widgetStyleValues[2],
                    widgetStyleValues[3],
                    widgetStyleValues[4],
                    widgetStyleValues[5],
                    widgetStyleValues[6],
                    widgetStyleValues[7]
            };
        }
    }

    @Override
    public void initView() {
        super.initView();
        clockFontContainer.setVisibility(View.GONE);
    }

    @Override
    public RemoteViews getRemoteViews() {
        return DayWidgetIMP.getRemoteViews(
                this, getLocationNow(),
                viewTypeValueNow, cardStyleValueNow, cardAlpha, textColorValueNow, textSize,
                hideSubtitle, subtitleDataValueNow
        );
    }

    @Override
    public String getSharedPreferencesName() {
        return getString(R.string.sp_widget_day_setting);
    }
}

File: app/src/main/java/org/breezyweather/remoteviews/config/DayWidgetConfigActivity.kt
/**
 * This file is part of Breezy Weather.
 *
 * Breezy Weather is free software: you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as published by the
 * Free Software Foundation, version 3 of the License.
 *
 * Breezy Weather is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
 * License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with Breezy Weather. If not, see <https://www.gnu.org/licenses/>.
 */

package org.breezyweather.remoteviews.config

import android.view.View
import android.widget.RemoteViews
import breezyweather.data.location.LocationRepository
import breezyweather.data.weather.WeatherRepository
import breezyweather.domain.location.model.Location
import dagger.hilt.android.AndroidEntryPoint
import org.breezyweather.R
import org.breezyweather.remoteviews.presenters.DayWidgetIMP
import org.breezyweather.sources.SourceManager
import javax.inject.Inject

/**
 * Day widget config activity.
 */
@AndroidEntryPoint
class DayWidgetConfigActivity : AbstractWidgetConfigActivity() {
    var locationNow: Location? = null
        private set

    @Inject
    lateinit var locationRepository: LocationRepository

    @Inject
    lateinit var weatherRepository: WeatherRepository

    @Inject
    lateinit var sourceManager: SourceManager

    override suspend fun initLocations() {
        val location = locationRepository.getFirstLocation(withParameters = false)
        locationNow = location?.copy(
            weather = weatherRepository.getWeatherByLocationId(
                location.formattedId,
                withDaily = true,
                withHourly = false,
                withMinutely = false,
                withAlerts = true // Custom subtitle
            )
        )
    }
    override fun initView() {
        super.initView()
        mViewTypeContainer?.visibility = View.VISIBLE
        mCardStyleContainer?.visibility = View.VISIBLE
        mCardAlphaContainer?.visibility = View.VISIBLE
        mHideSubtitleContainer?.visibility = View.VISIBLE
        mSubtitleDataContainer?.visibility = View.VISIBLE
        mTextColorContainer?.visibility = View.VISIBLE
        mTextSizeContainer?.visibility = View.VISIBLE
    }

    override fun updateWidgetView() {
        DayWidgetIMP.updateWidgetView(
            this,
            locationNow,
            locationNow?.let { location ->
                sourceManager.getPollenIndexSource((location.pollenSource ?: "").ifEmpty { location.weatherSource })
            }
        )
    }

    override val remoteViews: RemoteViews
        get() {
            return DayWidgetIMP.getRemoteViews(
                this,
                locationNow,
                viewTypeValueNow,
                cardStyleValueNow,
                cardAlpha,
                textColorValueNow,
                textSize,
                hideSubtitle,
                subtitleDataValueNow,
                locationNow?.let { location ->
                    sourceManager.getPollenIndexSource((location.pollenSource ?: "").ifEmpty { location.weatherSource })
                }
            )
        }

    override val configStoreName: String
        get() {
            return getString(R.string.sp_widget_day_setting)
        }
}
